Embed template into final build, fixed issue #54

dnauck opened the pull request:
* moved template from tpl to src folder
* adjusted links to template files in source and test configuration
* updated build process to create a to js converted and minified template that is added to the directives $templateCache
* updated the build process to provide a distribution with normal and minified versions of the directive with or without the included template (like angular-ui-bootstrap)
* added new dev dependencies to package.json
* adjust path to main file in bower.json

begriffs commented:
That was a significant amount of work, thank you! Can you also update the readme to reflect the templating change?

When I want to release new versions of the directive what process should I follow to ensure the template and everything is up to date (grunt tasks to run etc)?

dnauck commented:
I updates the readme and fixed some issue with the grunt tasks.

For the release: i added a new target called ```makeRelease``` that you can use the the old ```bump```task. It'll bump the version, test, build and commit the release.

Example:

```
$ grunt makeRelease --set-version 4.0.0
....
```

begriffs commented:
Nice, this looks great. One last thing: can you send a PR for the `gh-pages` branch too so that the demo page will work with the new template system?

dnauck commented:
Maybe its good to make a new release now for bower?

begriffs commented:
Before adding the official tag I was waiting for the problem with the demo to be resolved just as a last check that everything is OK.

begriffs commented:
OK, now that the demo is up and working I tried to make the release

```
$ grunt makeRelease --set-version 4.0.0 --dry-run
```

but it only bumped the version by a patch (to 3.1.2). I think the `bump-only` task needs a named argument like `major`, `minor`, or `patch`. I'd do something like `grunt makeRelease:major` but I don't know how to pass the argument down into `bump-only`. Thoughts?

dnauck commented:
```$ grunt makeRelease --setversion 4.0.0  --dry-run``` works without the dash between set and version (remove ```--dry-run``` for real release)

Why the task does not pass the arguments ```:major```, ```:minor```, ```:patch``` to the sub tasks .. i don't know. But i'll investigate.

begriffs commented:
That worked so much more cleanly than my old process for doing releases, thanks a lot. The version is now bumped to 4.0.0.

Now that I can specify a version directly I don't really need the major minor etc flags.
